aten/src/ATen/core/Type.h

```diff
@@ -45,64 +45,14 @@ struct Quantizer;
 static inline void noop_deleter(void*) {}
 
 enum class TypeID {
-  CPUBool,
-  CPUByte,
-  CPUChar,
-  CPUDouble,
-  CPUFloat,
-  CPUInt,
-  CPULong,
-  CPUShort,
-  CPUHalf,
-  CPUQInt8,
-  SparseCPUBool,
-  SparseCPUByte,
-  SparseCPUChar,
-  SparseCPUDouble,
-  SparseCPUFloat,
-  SparseCPUInt,
-  SparseCPULong,
-  SparseCPUShort,
-  SparseCPUQInt8,
-  MkldnnCPUFloat,
-  CUDABool,
-  CUDAByte,
-  CUDAChar,
-  CUDADouble,
-  CUDAFloat,
-  CUDAInt,
-  CUDALong,
-  CUDAShort,
-  CUDAHalf,
-  CUDAQInt8,
-  SparseCUDABool,
-  SparseCUDAByte,
-  SparseCUDAChar,
-  SparseCUDADouble,
-  SparseCUDAFloat,
-  SparseCUDAInt,
-  SparseCUDALong,
-  SparseCUDAShort,
-  SparseCUDAQInt8,
-  QuantizedCPUQInt8,
-  MSNPUBool,
-  MSNPUByte,
-  MSNPUChar,
-  MSNPUDouble,
-  MSNPUFloat,
-  MSNPUInt,
-  MSNPULong,
-  MSNPUShort,
-  MSNPUHalf,
-  XLABool,
-  XLAByte,
-  XLAChar,
-  XLADouble,
-  XLAFloat,
-  XLAInt,
-  XLALong,
-  XLAShort,
-  XLAHalf,
+  CPU,
+  SparseCPU,
+  MkldnnCPU,
+  CUDA,
+  SparseCUDA,
+  QuantizedCPU,
+  MSNPU,
+  XLA,
   CPUComplexFloat,
```

Comment:
what's the story with Complex btw? - why there are still switches based on scalar type?